plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -379,6 +379,7 @@ export class DefaultEntitiesCatalog implements EntitiesCatalog {
               facet.substring('metadata.labels.'.length)
             ];
           }
+          if (facet === entity.kind) return entity.metadata.name;
```

Comment:
This is absolutely not right, the semantics for asking for "Group" and getting back "my-team" is weird. You should remove this, and remember to remove the changeset line too.